pkg/util/proto/proto.go

```diff
@@ -96,3 +99,112 @@ func MustToStruct(message proto.Message) *structpb.Struct {
 	}
 	return str
 }
+
+type MergeFunction func(dst, src protoreflect.Message)
+type mergeOptions struct {
+	customMergeFn map[protoreflect.FullName]MergeFunction
+}
+type OptionFn func(options mergeOptions) mergeOptions
+
+func MergeFunctionOptionFn(name protoreflect.FullName, function MergeFunction) OptionFn {
+	return func(options mergeOptions) mergeOptions {
+		options.customMergeFn[name] = function
+		return options
+	}
+}
+
+// ReplaceMergeFn instead of merging all subfields one by one, takes src and set it to dest
+var ReplaceMergeFn MergeFunction = func(dst, src protoreflect.Message) {
+	dst.Range(func(fd protoreflect.FieldDescriptor, v protoreflect.Value) bool {
+		dst.Clear(fd)
+		return true
+	})
+	src.Range(func(fd protoreflect.FieldDescriptor, v protoreflect.Value) bool {
+		dst.Set(fd, v)
+		return true
+	})
+}
+
+func MergeForKuma(dst, src proto.Message) {
+	duration := &durationpb.Duration{}
+	Merge(dst, src, MergeFunctionOptionFn(duration.ProtoReflect().Descriptor().FullName(), ReplaceMergeFn))
+}
+
+// Merge Code of proto.Merge with modifications to support custom types
```

Comment:
If we are copying code in from somewhere else, we probably should preserve copyright and licenses?